core/src/main/java/org/apache/iceberg/BaseSnapshot.java

```diff
@@ -53,6 +55,7 @@ class BaseSnapshot implements Snapshot {
   private transient List<DeleteFile> addedDeleteFiles = null;
   private transient List<DeleteFile> removedDeleteFiles = null;
 
+  /** Tests only */
```

Comment:
I think there is an annotation that should be used instead, `VisibleForTesting`.